app-frontend/src/app/components/colorCorrectPane/colorCorrectPane.controller.js

```diff
@@ -65,9 +38,46 @@ export default class ColorCorrectPaneController {
      * @returns {null} null
      */
     onCorrectionChange(newCorrection) {
-        this.correction = newCorrection;
-        for (let layer of this.selectedLayers.values()) {
-            layer.colorCorrect(this.correction);
+        if (newCorrection) {
+            this.correction = newCorrection;
+            for (let layer of this.selectedLayers.values()) {
+                layer.colorCorrect(this.correction);
+            }
+            this.fetchHistogramData();
         }
     }
+
+    fetchHistogramData() {
+        this.$http.get(this.firstLayer.getHistogramURL()).then(
```

Comment:
IO interactions in Angular should be placed in a service of some sort. In this case, it might make sense to have `Layer` be responsible for getting its current histogram, or adding a new histogram service. Either way, the `$http` interaction should be moved.